tests/components/prometheus/test_init.py

```diff
@@ -87,6 +87,71 @@
 PROMETHEUS_PATH = "homeassistant.components.prometheus"
 
 
+@dataclass(frozen=True)
+class EntityTestMetricInfo:
+    """Class for all entity data in a single prometheus metric."""
+
+    metric_name: str
+    domain: str
+    friendly_name: str
+    object_id: str
+    metric_value: Any | None = None
+    mode: str | None = None
+    state: str | None = None
+    action: str | None = None
```

Comment:
As you noticed, not all metrics currently have these labels, and we could have metrics in the future that use some other labels.

I would advise against hardcoding them like this. Might be better to store them in a dict perhaps?

Reply:
I had liked the idea of typing, but if you'd prefer a dict then I'll go with that.